files/aws_cluster_autodiscover

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+set -o nounset
+
+# Gaffa tape script that uses AWS tags to pick an available ID/Index from a list of hostnames
+# This ID can be used to assign then pick the hostname and machine ID for services that require such a setting
+# such as Kafka or Zookeeper.
+#
+# NOTE In the future we should replace this with Consul.
+#
+# Usage:
+# aws_cluster_autodiscover [Max number of machines in cluster] [AWS hostedzone ID] "[AWS ec2 describe-instances filter]" [Name of tag used to store id]
+# eg.
+# ./bin/aws_cluster_autodiscover "01.zookeeper.app.internal,02.zookeeper.app.internal,03.zookeeper.app.internal" "Z8LHX0FIHPY9F" "Name=tag:Environment,Values=dev Name=tag:Role,Values=zookeeper" "ZookeeperID"
+#
+# Returns:
+# Json array detaling the claimed ID and hostname
+# eg.
+# {"id": "1", "id_index0": "0", "hostname": "01.zookeeper.app.internal", "info": "Machine had no ID, claimed ID and R53 entry" }
+
+function die {
+	echo -e "\033[1;31mError: $1\033[0m"
+	exit 1
+}
+
+function output {
+	local -i CLAIMED_ID=$1
+	local CLAIMED_HOST=$2
+	local INFO=$3
+	local -i CLAIMED_ID_INDEX0=$((CLAIMED_ID - 1 ))
+	echo "{\"id\": \"${CLAIMED_ID}\", \"id_index0\": \"${CLAIMED_ID_INDEX0}\", \"hostname\": \"${CLAIMED_HOST}\", \"info\": \"${INFO}\" }"
+	exit 0
+}
+
+function main {
+	# Arguments
+	if [ "$#" -ne 4 ]; then
+		die "This script takes 4 parameters, only $# given"
+	fi
+	local HOSTS=(${1//,/ })
+	local HOSTED_ZONE_ID=$2
+	local LOOKUP_FILTER=$3
+	local ID_TAG_NAME=$4
+
+	# Initialise vars before subshell since local always returns 0
+	local MAX_HOSTS=${#HOSTS[@]}
+	local CLAIMED_ID=""
+	local CLAIMED_HOST=""
+	local INSTANCE_ID=""
+	local INSTANCE_IP=""
+	local INSTANCE_REGION=""
+	local AVAILABLE_IDS=( )
+	local CLAIMED_IDS=( )
+	local UNCLAIMED_IDS=( )
+
+	INSTANCE_ID=$( curl -f -s http://169.254.169.254/latest/meta-data/instance-id/ )
+	INSTANCE_IP=$( curl -f -s http://169.254.169.254/latest/meta-data/local-ipv4/ )
+	INSTANCE_REGION=$(curl -s -f http://169.254.169.254/latest/meta-data/placement/availability-zone | sed '$s/.$//')
```

Comment:
(minor) curl flags order reversed compared to previous lines.
Is the sed syntax working without an ending separator? I'd suggest expliciting, e.g., sed -e 's!s/.!//!'